src/TestSuite/Constraint/EventFiredWith.php

```diff
@@ -0,0 +1,105 @@
+<?php
+namespace Cake\TestSuite\Constraint;
+
+use Cake\Event\Event;
+use Cake\Event\EventManager;
+use PHPUnit_Framework_AssertionFailedError;
+use PHPUnit_Framework_Constraint;
+
+/**
+ * EventFiredWith constraint
+ *
+ * Another glorified in_array check
+ */
+class EventFiredWith extends PHPUnit_Framework_Constraint
+{
+    /**
+     * Array of fired events
+     *
+     * @var EventManager
+     */
+    protected $_eventManager;
+
+    /**
+     * Event data key
+     *
+     * @var string
+     */
+    protected $_dataKey;
+
+    /**
+     * Event data value
+     *
+     * @var string
+     */
+    protected $_dataValue;
+
+    /**
+     * Constructor
+     *
+     * @param EventManager $eventManager Event manager to check
+     * @param string $dataKey Data key
+     * @param string $dataValue Data value
+     */
+    public function __construct($eventManager, $dataKey, $dataValue)
+    {
+        parent::__construct();
+        $this->_eventManager = $eventManager;
+        $this->_dataKey = $dataKey;
+        $this->_dataValue = $dataValue;
+
+        if ($this->_eventManager->getEventList() === null) {
+            throw new PHPUnit_Framework_AssertionFailedError('The event manager you are asserting against is not configured to track events.');
```

Comment:
There's no current way to check the current state of `trackEvents` so I left it off. Setting an event list automatically sets them to be tracked, so it seems the developer needs to deliberately `trackEvents(false)` for it to fail, in which case it just won't show in the list and the assertion would fail anyway. Adding the ability to check the track status could be part of improving EventList in a separate PR, which I'm happy to do.